daffodil-lib/src/main/scala/org/apache/daffodil/api/ParseUnparsePolicyTunable.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.api
+
+import org.apache.daffodil.exceptions.ThrowsSDE
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.schema.annotation.props.Enum
+
+/**
+ * Specifies to compile DFDL schemas with regards to parse vs unparse
+ * capabilities.
+ *
+ * ParseOnly:
+ *  Compile for only parsing
+ *
+ * UnparseOnly:
+ *   Compile for only unparsing
+ *
+ * Both:
+ *   Compile for both parsing and unparsing
+ *
+ * FromRoot:
+ *   Use the dfdl:parseUnparsePolicy from the root element to determine how to
+ *   compile the schema
+ */
+sealed trait ParseUnparsePolicyTunable extends ParseUnparsePolicyTunable.Value
```

Comment:
We could generate this trait also yes?  Or is there some reason this and the next one UnqualifiedPathStepPolicy need special treatment? 

If there is a reason, please add comments to code to that effect.

Reply:
Yeah, pretty sure this can be autogenerated too. Will add.